modules/manually-maintained-credentials-upgrade.adoc

```diff
@@ -5,9 +5,22 @@
 // * installing/installing_gcp/manually-creating-iam-gcp.adoc
 
 [id="manually-maintained-credentials-upgrade_{context}"]
-== Upgrades
+= Upgrading clusters with manually maintained credentials
 
-In a future release, improvements to the Cloud Credential Operator will prevent
-situations where a user might enter an upgrade that will fail because their
-manually maintained credentials have not been updated to match the
-`CredentialsRequests` in the upcoming release image.
+If permissions change in a future release, the Cloud Credential Operator (CCO)
```

Comment:
@dgoodwin how best to document this? If there are brand new credentials, CCO can mark itself upgradable=False (all that work you did in those recent PRs), but CCO doesn't really have a way to handle if ExistingCredReqA has new permissions requirements from one version to the next (eg. image-registry now needs some encryption-related permissions that weren't previously required). The best we can presently do is hope that the secret containing the credentials is good enough. Given the limitations with permissions simulations, I'm reluctant to rely on the results of permissions simulations solely to be able to reliably be able to flag "ExistingCredReqA needs updating".

Reply:
@dgoodwin & @joelddiaz  -
Circling back on this: with the step order fixed, I think this comment is the only outstanding issue on this PR. LMK what y'all think we ought to say about it and I'll be happy to get this one wrapped up.

Reply:
This is an interesting one, in manual mode we don't do *any* of the policy checking to make sure it's in sync with what's in the CredentialsRequest IIRC. We still should have our read-only cred, so in theory we could change that and start comparing policy and reporting problems.

If we did do that work, then we would want to update the upgradeable false to not just look for existence of a secret but also try to verify the permissions are correct on it. 

So today just a secret check, which is consistent with manual mode behavior anyhow, it's on you to make sure the permissions are right. We could possibly improve this in future.

Reply:
No policy simulation required I hope.

Reply:
So I guess we need another section here, something like:

In addition to creating new credentials for the release image you wish to upgrade to, you must also compare the required permissions for existing credentials to accommodate any pre-existing components which require new permissions.

This would flow into the last line about seeing the manual IAM instructions.